src/core/Akka.Remote/Transport/Helios/HeliosHelpers.cs

```diff
@@ -21,224 +22,50 @@
 
 namespace Akka.Remote.Transport.Helios
 {
-    /// <summary>
-    /// INTERNAL API
-    /// 
-    /// Abstract base class for adapting Helios <see cref="IConnection"/> objects to work with Akka.Remote.Transports
-    /// </summary>
-    internal abstract class HeliosHelpers : IConnection
+    internal abstract class CommonHandlers : ChannelHandlerAdapter
```

Comment:
Apologies for the unreadable diff here - Helios 2.0 requires much less code for its transport handlers due to the pipeline model. Basically we hand the yet-to-be-serialized message over to protobuf and decrease the reference count on the buffer.